### PR TITLE
USAGOV-978: Sign our images with Docker Content Trust

### PR DESCRIPTION
**Pull request by akf**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-978

## Description

Adds a signing step for the Docker images we build and publish to Docker Hub, and use Docker Content Trust (by having  DOCKER_CONTENT_TRUST=1 in env) when pulling base images where possible and also when pulling our own images in order to retrieve their image hashes. 

I learned that the base image we pull for the cms app is signed, but the owasp/modsecurity image we pull for the waf is not (boo!). So I looked up the image hash for the owasp/modsecurity image we use and supplied it instead. 

Removes ${NODE_BASE} in the node image and replaces it with simply "node," because arguments in the FROM line of an  image file triggers a bug with Docker Content Trust. Happily, we no longer need that variable, because the "node" image is now multi-architecture.

Also removes a bunch of conditional sudo-ing, which I now suspect we were doing simply because it seemed to be resolving a problem that looked like "lack of permission to connect to a socket" but was really "can't use arguments in the FROM line of an image." (Sudo did make the problem go away, by eliminating the environment variable DOCKER_CONTENT_TRUST that I know we did have set in CircleCI in the past.) See the Jira ticket for background on that problem. 

Explicitly turns off Docker Content Trust when using snyk:snyk images, because Snyk doesn't sign those. (Why???) 

The most recent logs from bulding this branch can be inspected here: 
https://app.circleci.com/pipelines/github/usagov/usagov-2021/5586/workflows/6c9eb038-e5bf-4afd-b2bc-fcc1e2314a1a/jobs/9984

There is one new script that isn't actually used: bin/cloudgov/container-sign. I tested it and it was working, but once I got the scripts working with DOCKER_CONTENT_TRUST set to 1, I realized signing happens automatically when the container is pushed.

Finally, my last couple of updates triggered lint errors in phpcs from completely unrelated code. Apparently "use" statements are supposed to be alphabetical now. Seems like a new error. Anyway, I fixed those to get linting to pass.

## Checklist for the Developer

- [x] A link to the JIRA ticket has been added above.
- [ ] The JIRA ticket identifies the desired result of this change.
- [ ] The JIRA ticket contains clear acceptance criteria.
- [ ] The JIRA ticket contains clear testing steps.
- [ ] The JIRA ticket contains a description of "Done".
- [ ] Any preparation/installation/update steps required for this code to work properly (drush commands, scripts, configuration updates) are documented in the ticket.

## Checklist for the Peer Reviewers
- [ ] The branch name of this PR matches the project standards.
- [ ] QA steps are followed and any changes to process are updated in the ticket.
- [ ] Code Standards are followed, there are no bad practices in use.
- [ ] Config changes (if any) include only necessary changes.
- [ ] No errors in Drupal or Client Browser.
- [ ] Code has been tested locally (if possible).
- [ ] Following AC and Testing Steps verify changes work as expected.
- [ ] There are no known side-effects outside of expected behavior.

### CODE REVIEW

**Comment by akf** [comment]
Just realized I have an error here; converting to draft.

File: .circleci/config.yml [hunk 1]
```diff
@@ -59,10 +55,8 @@ commands:
             if aws s3 cp s3://${S3_BUCKET}/${DOCKERUSER}/${DOCKERREPO}/waf-<< pipeline.number >> ./waf-image-digest; then export WAF_DIGEST="@"$(cat ./waf-image-digest); fi
             export CMS_DIGEST=$CMS_DIGEST
             export WAF_DIGEST=$WAF_DIGEST
-            echo 'export CMS_DIGEST='"$CMS_DIGEST" | $SUDO tee -a $BASH_ENV
-            echo 'export WAF_DIGEST='"$WAF_DIGEST" | $SUDO tee -a $BASH_ENV
-            # docker pull ${DOCKERUSER}/${DOCKERREPO}:cms-<< pipeline.number >>${CMS_DIGEST}
-            # docker pull ${DOCKERUSER}/${DOCKERREPO}:waf-<< pipeline.number >>${WAF_DIGEST}
+            echo 'export CMS_DIGEST='"$CMS_DIGEST" | tee -a $BASH_ENV
```
Comment (ghost) [inline .circleci/config.yml:58]: Do we know what $BASH_ENV expands to in this context?

I was trying to figure out what this did, and none of my environments have this env var set.

**Review by ghost — APPROVED** [review]
Looks ok.   I did not try to run container-sign, as you mentioned it is not needed.

File: .circleci/config.yml (hunk 1 above)
Comment (akf) [reply .circleci/config.yml:58]: It's set to /home/circleci/project/env.local in .circleci/config.yml
I just looked this up, and immediately realized I'd search for it before and forgotten, so ...

**Pull request merged**